students/brgalloway/Lesson_3/mailroom_part_1.py

```diff
@@ -43,66 +66,28 @@ def generate_report(donors_list):
     print("-" * 70)
     for i in sorted_list:
         print(f"{i[0]:<20}${i[1]:>14.2f}{i[2]:^18}${i[3]:>12.2f}".format())
-    return user_prompt(donors_list)
-
-# this function only finds the users 
-# in the donor database. if the user
-# is not found it will then pass that 
-# value to the send_thankyou letter 
-# where new users are added and recorded
-def find_donor():
-    fullname = input("Enter the full name of the donor or type list to display names: ")
-    # inner function for displaying list
-    # and recalling the send_thankyou() function
-    def list_names():
-        donors_list.sort()
-        for i in range(len(donors_list)):
-            print(donors_list[i][0])
-        return find_donor()
 
-    while fullname != "quit":
-    # check what the user has entered
-        if fullname == "list":
-            return list_names()
-        elif fullname in str(donors_list):
-            print("Selecting Donor " + fullname)
-            for i in range(len(donors_list)):
-                if fullname in str(donors_list[i]):
-                    fullname = donors_list[i][0]  
-            return send_thankyou(fullname)
-        elif fullname == "quit":
-            return user_prompt()
-    # Else user not in database send new 
-    # user to the send_thankyou function
-        else:
-            donors_list.append([fullname])
-            for i in range(len(donors_list)):
-                if fullname in str(donors_list[i]):
-                    fullname = donors_list[i][0]
-            return send_thankyou(fullname,donors_list)
-    
 # This function sends the formatted email
 # records donation amounts and adds new users 
 # and their donaitons to the database
-def send_thankyou(fullname, donors_list=donors_list, times_donated=0, average_donation=0):
-    donation_amount = float(input(f"Donation amount: "))
-    for i in range(len(donors_list)):
-        if fullname in str(donors_list[i]):
-            if len(donors_list[i]) == 1:
-                donors_list[i].extend((donation_amount, 0, 0))
-            else:
-                add_donation = donors_list[i][1] + donation_amount
-                donors_list[i].pop(1)
-                donors_list[i].insert(1,add_donation)
-                
+def send_thankyou(fullname,donors_list=donors_list):
+    donation_amount = float(input("Donation amount: "))
+    for donor in donors_list:
+        if fullname == donor[0]:
+            donor[1] = donor[1] + donation_amount
+            donor[2] = donor[2] + 1
+            donor[3] = donor[1] / donor[2]
+            donors_list[donors_list.index(donor)] = [donor[0], donor[1], donor[2], donor[3]]
```

Comment:
you don't need this line, the modifications you applied in lines 77, 78, 79 have already modified your list.